be/src/exprs/math_functions.cpp

```diff
@@ -768,6 +768,17 @@ StatusOr<ColumnPtr> MathFunctions::cosine_similarity(FunctionContext* context, c
                 fmt::format("cosine_similarity does not support null values. {} array has null value.",
                             base->has_null() ? "base" : "target"));
     }
+    if (base->is_constant()) {
+        auto* const_column = down_cast<const ConstColumn*>(base);
+        const_column->data_column()->assign(base->size(), 0);
+        base = const_column->data_column().get();
+    }
+    if (target->is_constant()) {
+        auto* const_column = down_cast<const ConstColumn*>(target);
+        const_column->data_column()->assign(target->size(), 0);
+        target = const_column->data_column().get();
+    }
```

Comment:
use unfold_const() function?

Reply:
here must not be `only_null`, so I only check `is_constant`